packages/devextreme/js/__internal/ui/speech_to_text/speech_to_text.ts

```diff
@@ -0,0 +1,261 @@
+
+  private _actions!: SpeechToTextActions;
+
+  _getDefaultOptions(): SpeechToTextProperties {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+    return extend(super._getDefaultOptions(), {
+      stylingMode: 'contained',
+      type: 'default',
+      startIcon: DEFAULT_INITIAL_ICON,
+      stopIcon: DEFAULT_LISTENING_ICON,
+      startText: '',
+      stopText: '',
+      customSpeechRecognizer: {
+        enabled: false,
+        isListening: false,
+      },
+      onStartClick: undefined,
+      onStopClick: undefined,
+      onResult: undefined,
+      onError: undefined,
+    });
+  }
+
+  _init(): void {
+    this._actions = {};
+    super._init();
+    this._handleCustomEngineState();
+    this._createActions();
+  }
+
+  _initMarkup(): void {
+    super._initMarkup();
+    this.$element().addClass(SPEECH_TO_TEXT_CLASS);
+    this._renderButton();
+    this._updateButtonState();
+  }
+
+  private _createActions(): void {
+    ACTIONS.forEach((action) => {
+      this._setAction(action);
+    });
+  }
+
+  private _setAction(action: keyof SpeechToTextActions): void {
+    this._actions[action] = this._createActionByOption(action, {
+      excludeValidators: ['disabled', 'readOnly'],
+    }) || noop;
+  }
+
+  private _renderButton(): void {
+    this._button = this._createComponent<Button, ButtonProperties>(
+      this.$element(),
+      Button,
+      this._getButtonOptions(),
+    );
+  }
+
+  private _getButtonOptions(): ButtonProperties {
+    const {
+      activeStateEnabled,
+      focusStateEnabled,
+      hoverStateEnabled,
+      stylingMode,
+      type,
+      disabled,
+      width,
+      height,
+    } = this.option();
+
+    return {
+      stylingMode,
+      type,
+      disabled,
+      width,
+      height,
+      activeStateEnabled,
+      focusStateEnabled,
+      hoverStateEnabled,
+      icon: this._getCurrentIcon(),
+      text: this._getCurrentText(),
+      onClick: (e: ClickEvent): void => {
+        this._buttonClickHandler(e);
+      },
+    };
+  }
+
+  private _getCurrentIcon(): string | undefined {
+    const { startIcon, stopIcon } = this.option();
+
+    return this._isListening() ? stopIcon : startIcon;
+  }
+
+  private _getCurrentText(): string {
+    const { startText, stopText } = this.option();
+
+    return this._isListening()
+      ? stopText ?? ''
+      : startText ?? '';
+  }
+
+  private _buttonClickHandler(e: ClickEvent): void {
+    if (this._state === SpeechToTextState.DISABLED) {
+      return;
+    }
+
+    if (this._isListening()) {
+      this._handleStopClick(e);
+    } else {
+      this._handleStartClick(e);
+    }
+  }
+
+  private _handleStartClick(e: ClickEvent): void {
+    const { customSpeechRecognizer } = this.option();
+
+    if (!customSpeechRecognizer?.enabled) {
+      this._setState(SpeechToTextState.LISTENING);
+    }
+
+    this._actions.onStartClick?.({
+      component: this,
+      element: this.$element().get(0) as HTMLElement,
```

Comment:
[blocking]
nope, it's still should be a jQuery element if jquery is imported to the page
```suggestion
      element: this.$element(),
```

Reply:
Hmm, yes, you're right, we actually have a utility for selecting which element to return: `getPublicElement`. I'll use it, it automatically resolves the element type.

Reply:
Used `this.element()`: [9e2a7752de838dac87a3a6aa040186168a64c19c]